### ISSUE TITLE
Incompatibility with doctrine/dbal 3

### ISSUE THREAD

**Issue by jderusse** [ScientaNL/DoctrineJsonFunctions#74]
When upgraded to doctrine/dbal 3, an exception `Attempted to load class "DBALException" from namespace "Doctrine\DBAL".`
Because this class does not exist anymore.

This classe is used by `PostgresqlJsonOperatorFunctionNode`, and check if the Platform is an instance of `PostgreSqlPlatform`, which is wrong in doctrine/dbal 3, the instance is now a `PostgreSQL100Platform`

This happens, because the compose.json file of `scienta/doctrine-json-functions` does not have any constraint on 3rd party library.
I think you should add constraints in the `require` section. Or, if you want to keep this section optional, you should add the constraint in the `conflict` section to prevent people from using versions that are not tested.

**Comment by webdevilopers** [comment]
Thanks @jderusse for opening this issue. I recently had to update a sonata-admin project and needed "JSON_GET_TEXT" in Postgresql for a Doctrine ORM "json" column.

Thank you @piotaixr for the PR a few days ago.

How are the chances the PR gets merged soon? Or can the feature branch be used `dbal3.0`?
https://github.com/piotaixr/DoctrineJsonFunctions/tree/feature/dbal3.0

**Comment by florentdestremau** [comment]
For those like me waiting for the merge of the PR from @piotaixr you can already use it with this config in your composer.json:

```
    "require": [
        "scienta/doctrine-json-functions": "dev-feature/dbal3.0",
    ]
    "repositories": [
        {
            "type": "vcs",
            "url":  "https://github.com/piotaixr/DoctrineJsonFunctions"
        }
    ],
```

**Comment by webmasterMeyers** [comment]
+1 currently using piotaixr/DoctrineJsonFunctions to get around this

**Comment by wouter-toppy** [comment]
Any update on this? can this be released?

**Comment by Hikariii** [comment]
https://github.com/ScientaNL/DoctrineJsonFunctions/pull/76 will be tested and released the coming week under version `5.0`
